### PR TITLE
Add classroom and teacher record edit feature

### PR DESCRIPTION
**Pull request by nkelava**
This PR will add edit feature for classroom and teacher records. First, I added icons for edit, save and cancel operations. When user enters the page he can see edit icon next to the delete icon by default now. On edit icon click, two icons replace the edit icon and these icons are save and cancel icons. Also, input elements can be seen on record fields that are editable. Fields that we don't want any user to be able to change can be disabled for editing by setting field's isEditable property to false. If the user clicks save icon then the data is updated but if he clicks cancel icon then editing is canceled. In case of cancellation any new values are lost and everything is back to old data.

### CODE REVIEW

File: frontend/src/api/classrooms.js [hunk 1]
```diff
@@ -4,15 +4,26 @@ const baseUrl = '/api/classrooms';
 
 const getAllClassrooms = () => fetcher(`${baseUrl}/`);
 
-const addClassroom = ({ name, capacity }) =>
+const addClassroom = ({ name, capacity }) => {
   fetcher(`${baseUrl}`, {
```
Comment (abasic) [inline frontend/src/api/classrooms.js]: Shouldn't there be a return here?

File: frontend/src/api/teachers.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const baseUrl = '/api/teachers';
 
 const getAllTeachers = () => fetcher(`${baseUrl}/`);
 
-const addTeacher = ({ teacherCode, firstName, lastName }) =>
+const addTeacher = ({ teacherCode, firstName, lastName }) => {
   fetcher(`${baseUrl}`, {
```
Comment (abasic) [inline frontend/src/api/teachers.js]: Shouldn't there be a return here?

File: frontend/src/components/Classroom/ClassroomList.vue [hunk 3]
```diff
@@ -10,18 +13,51 @@ const props = defineProps({
...
+];
+const classroomEditingId = ref(null);
 
 const emit = defineEmits(['reload']);
 const filteredClassrooms = computed(() => {
   const searchQueryLowercased = searchQuery.value.toLowerCase();
   const filtered = props.classrooms.filter(classroom => classroom.name.toLowerCase().includes(searchQueryLowercased));
   return filtered;
 });
+const isToEdit = classroomId => {
```
Comment (abasic) [inline frontend/src/components/Classroom/ClassroomList.vue]: Nitpick: this name seems to me to be a little convoluted and difficult to understand. Should it be something like 'isBeingEdited' or something like that?

File: frontend/src/components/Teacher/TeacherList.vue [hunk 4]
```diff
@@ -47,15 +88,35 @@ const handleDelete = async teacherId => {
...
+              <span v-if="isToEdit(teacher.id) && field.isEditable">
+                <input v-model="teacher[field.property]" type="text" class="rsprd-input--edit" />
+              </span>
+              <span v-else>
+                {{ teacher[field.property] }}
+              </span>
+            </td>
             <td class="rsprd-table__cell">
-              <button @click="handleDelete(teacher.id)" class="rsprd-btn rsprd-btn--delete">
+              <span v-if="isToEdit(teacher.id)">
```
Comment (abasic) [inline frontend/src/components/Teacher/TeacherList.vue]: Nitpick: `div` is a more appropriate element to wrap groups of other elements. `span` should be more for text.

**Review by bdeak4 — COMMENTED** [review]
Looks good! I'll just implement sort for these lists so items don't jump on edit in another PR

**Review by abasic — DISMISSED** [review]

**Review by bdeak4 — APPROVED** [review]

**Pull request merged**